scipy/stats/_fit.py

```diff
@@ -1104,6 +1107,7 @@ def rvs(size):
     # Define statistic
     fit_fun = _get_fit_fun(dist, data, guessed_rfd_params, fixed_rfd_params)
     compare_fun = _compare_dict[statistic]
+    alternative = getattr(compare_fun, 'alternative', 'greater')
```

Comment:
this line is causing some errors in the check matrix: 
https://github.com/scipy/scipy/actions/runs/4337818594/jobs/7574077614#step:12:80

From what I gather the `_compare_dict` is a map from string literals to functions, 
https://github.com/scipy/scipy/blob/main/scipy/stats/_fit.py#L1228
so I don't think the `getattr()` method is going to work and the functions themselves simply return a number (usually floats), e.g.
https://github.com/scipy/scipy/blob/main/scipy/stats/_fit.py#L1192

I don't have the full context here but perhaps the string-literal `'alternative'` should be something else?

Reply:
MyPy is complaining, but there is not a real error. A solution is to silence mypy https://github.com/scipy/scipy/pull/18102#pullrequestreview-1325282709.

Reply:
got it, agree